common/http-api-client/src/lib.rs

```diff
@@ -1048,8 +1086,27 @@ impl ApiClientCore for Client {
                 .build()
                 .map_err(HttpClientError::reqwest_client_build_error)?;
             self.apply_hosts_to_req(&mut req);
+            let url = Url::parse(req.url().as_str()).unwrap();
+
+            // try an explicit DNS resolution - if successful then it will be in cache when reqwest
+            // goes to execute the request. If failure then we get to handle the DNS lookup error.
             #[cfg(not(target_arch = "wasm32"))]
-            let url = req.url().clone();
+            if self.using_secure_dns
+                && let Some(hostname) = req.url().domain()
+                && let Err(err) = HickoryDnsResolver::default().resolve_str(hostname).await
```

Comment:
This `default` is non-obviously delegating to the shared resolver, maybe just add a little blurb for future us